### ISSUE TITLE
FC4-03 | `consensusThreshold` Can Not Be Modified

### ISSUE THREAD

**Issue by Brando753** [Sifchain/sifnode#3152]
This appears to still be an issue in the codebase, recommend peggy team huddle to discuss. Affected files Oracle.sol L49 and consensusNeeded.go L13.

# Description
In the `smart-contracts/contracts/Oracle` contract, after the contract is initialized, `consensusThreshold` cannot be modified.

```solidity
  function _initialize(
      address _operator,
      uint256 _consensusThreshold,
      address[] memory _initValidators,
      uint256[] memory _initPowers
    ) internal {
      require(!_initialized, "Initialized");
      require(_consensusThreshold > 0, "Consensus threshold must be positive.");
      require(_consensusThreshold <= 100, "Invalid consensus threshold.");
      operator = _operator;
      consensusThreshold = _consensusThreshold;
      _initialized = true;
      Valset._initialize(_operator, _initValidators, _initPowers);
    }
```
`consensusThreshold` represents the consensus rate, which has the same meaning as `consensusNeeded` in the Sifchain.
Assuming that `sumPower` represents the sum of validators' power who participated in validating the transaction, `totalPower` represents the sum of all validators' power. The cross-chain transaction from the Sifchain to the Ethereum chain needs to meet:
Ethereum chain:

  `sumPower` / `totalPower` >= `consensusThreshold`

Sifchain:

  `sumPower` / `totalPower` >= `consensusNeeded` 

Thus, the value of `consensusThreshold` in the Ethereum chain needs to be consistent with the value of `consensusNeeded` in the Sifchain. However, the `consensusNeeded` in the Sifchain can be modified. If `consensusNeeded < consensusThreshold`, the cross-chain transactions from the Sifchain to the Ethereum chain can never be executed.

# Recommendation
Recommend adding a privileged function that modifies consensusThreshold in contract Oracle, so that it can keep consensusThreshold consistent with consensusNeeded.

**Comment by Brando753** [comment]
Note: Both sifchain and smart contracts set a limit of 1-100 as required thresholds

**Comment by smartyalgo** [comment]
I think this is a legitimate issue, and could cause the bridge to halt

This concerns the flow from Sifchain to EVM
tl;dr 
When consensusNeeded (on sifchain) is higher than consensusThreshold (on CosmosBridge.sol), 
If a tx doesnt pass consensusNeeded, Sifchain wont emit event for relayer to relay that tx, even if it passes consensusThreshold. From EVM's perspective, that tx is stuck. 

Will post PR soon

**Comment by smartyalgo** [comment]
PR: https://github.com/Sifchain/sifnode/pull/3194
